## `Example`

    • Eagle Outlier is a chain of stores specializing in outdoor apparel and campiagian gear.They are considered a promotion  that involves mailing discount coupouns to all their credit card customers.This promotion wil be considered a success if more than 10% of those receiving the coupons use them.Before going national with the promotion ,coupons were sent to a sample of 100 credit cards customers.
     a.Develope hypothesis that can be used to test whether the population propotion of those who will use the coupons is    
       sufficient to go national.
     b.The file Eagle contains the sample data. Develope a point estimate of the population propotion.
     c.Use α .05 to conduct your hypothesis test. Should Eagle go national with the promotion?

In [1]:
import pandas as pd

eagle = pd.read_csv('E:\\python\\Eagle.csv')

In [6]:
eagle.describe()

Used_Coupon
count          100
unique           2
top             No
freq            87

In [3]:
Yes = eagle.groupby('Used_Coupon')
eagle_Yes = Yes.get_group('Yes')

In [4]:
eagle_Yes

Used_Coupon
3          Yes
6          Yes
24         Yes
27         Yes
36         Yes
49         Yes
50         Yes
52         Yes
68         Yes
73         Yes
90         Yes
92         Yes
93         Yes

In [5]:
eagle_Yes.describe()

Used_Coupon
count           13
unique           1
top            Yes
freq            13

In [7]:
import math
a = 0.13 -0.1
b = math.sqrt(0.1*(1-0.1)/100)
z = a/b

In [8]:
z

0.9999999999999999

    Population Propotion
    • H0 : P less than eqaul to 0.10
    • H1 : P>0.10 #population proportion is greater than 10%.

In [54]:
from scipy.stats import norm
pbar = 13/100 #sample propotion
pzero = 0.10 #hypothesised value
n =100
mu,sigma = 0,1

z1 = pbar - pzero
z2 = (pzero*(1-pzero)/n)**0.5

z13 = z1/z2
#The are on the left side of STD, NOR

p13 = norm.cdf(z13,mu,sigma)
#since it is upper tale test
print('Upper tale P value : %2.4f' %(1-p13))

Upper tale P value : 0.1587


    p-value 0.1587 is greater than 0.05. We accept H0 and hence should not go for National

In [49]:
from scipy.stats import norm
pbar = 14/100 #sample propotion
pzero = 0.10 #hypothesised value
n =100
mu,sigma = 0,1

z1 = pbar - pzero
z2 = (pzero*(1-pzero)/n)**0.5

z14 = z1/z2
#The are on the left side of STD, NOR

p14 = norm.cdf(z14,mu,sigma)
#since it is upper tale test
print('Upper tale P value : %2.4f' %(1-p14))

Upper tale P value : 0.0912


In [50]:
from scipy.stats import norm
pbar = 15/100 #sample propotion
pzero = 0.10 #hypothesised value
n =100
mu,sigma = 0,1

z1 = pbar - pzero
z2 = (pzero*(1-pzero)/n)**0.5

z15 = z1/z2
#The are on the left side of STD, NOR

p15 = norm.cdf(z15,mu,sigma)
#since it is upper tale test
print('Upper tale P value : %2.4f' %(1-p15))

Upper tale P value : 0.0478


In [51]:
from scipy.stats import norm
pbar = 16/100 #sample propotion
pzero = 0.10 #hypothesised value
n =100
mu,sigma = 0,1

z1 = pbar - pzero
z2 = (pzero*(1-pzero)/n)**0.5

z16 = z1/z2
#The are on the left side of STD, NOR

p16 = norm.cdf(z16,mu,sigma)
#since it is upper tail test
print('Upper tale P value : %2.4f' %(1-p16))

Upper tale P value : 0.0228


In [55]:
Ptable = pd.DataFrame({'Z':[z13,z14,z15,z16],'P':[1-p13,1-p14,1-p15,1-p16] ,'Accept': ['H0','H0','H1','H1']},index = [13,14,15,16])

In [56]:
Ptable

Z         P Accept
13  1.000000  0.158655     H0
14  1.333333  0.091211     H0
15  1.666667  0.047790     H1
16  2.000000  0.022750     H1

In [5]:
from scipy.stats import norm
import pandas as pd

def propotion(pro):
    pbar = pro/100
    pzero = 0.10 #hypothesised value
    n =100
    mu,sigma = 0,1
    
    z1 = pbar - pzero
    z2 = (pzero*(1-pzero)/n)**0.5

    z = z1/z2
    
    #The are on the left side of STD, NOR
    p = norm.cdf(z,mu,sigma)

    data = dict()
    data['pbar'] = pro
    data['z'] = z
    data['Value of p']= 1-p
    
    if (1 -p ) < 0.05:
        data['Accept'] = 'H1'
    else :
        data['Accept'] = 'H0'
    
    return data 


df100 = pd.DataFrame({'pbar':[],'z':[],'Value of p': [],'Accept': []})
for pro in range(13,20):
    df100= df100.append(propotion(pro),ignore_index = True)
    
    


In [6]:
df100

pbar         z  Value of p Accept
0  13.0  1.000000    0.158655     H0
1  14.0  1.333333    0.091211     H0
2  15.0  1.666667    0.047790     H1
3  16.0  2.000000    0.022750     H1
4  17.0  2.333333    0.009815     H1
5  18.0  2.666667    0.003830     H1
6  19.0  3.000000    0.001350     H1

### `Drowsy.csv`

     In a cover story, BusinessWeek published information about sleep habits of Americans (BusinessWeek, January 26, 2004). The article noted that sleep deprivation causes a number of problems, including highway deaths. Fifty-one percent of adult drivers admit to driving while drowsy. A researcher hypothesized that this issue was an even bigger problem for night shift workers.
    a. Formulate the hypotheses that can be used to help determine whether more than 51% of the population of night shift 
       workers admit to driving while drowsy.
    b. A sample of 500 night shift workers identified those who admitted to driving while drowsy. What is the sample 
       proportion? What is the p-value?
    c. At α = 0.01, what is your conclusion?

In [2]:
import pandas as pd
drowsy = pd.read_csv('E://python//Drowsy.csv')

In [3]:
drowsy

Drowsy
0      Yes
1      Yes
2       No
3      Yes
4       No
..     ...
495     No
496    Yes
497    Yes
498    Yes
499     No

[500 rows x 1 columns]

In [5]:
Yes = drowsy.groupby('Drowsy')
Drowsy_Yes = Yes.get_group('Yes')

In [23]:
Drowsy_Yes

Drowsy
0      Yes
1      Yes
3      Yes
5      Yes
7      Yes
..     ...
490    Yes
491    Yes
496    Yes
497    Yes
498    Yes

[232 rows x 1 columns]

In [6]:
Drowsy_Yes.count()

Drowsy    232
dtype: int64

In [24]:
from scipy.stats import norm
pbar = 232/500 #sample propotion
pzero = 0.51 #hypothesised value
n =500
mu,sigma = 0,1

z1 = pbar - pzero
z2 = (pzero*(1-pzero)/n)**0.5

z = z1/z2
#The are on the left side of STD, NOR

p = norm.cdf(z,mu,sigma)
#since it is upper tale test
print('Z : %2.4f'%z)
print('Upper tale P value : %2.4f' %(1-p))

Z : -2.0576
Upper tale P value : 0.9802


     p-value 0.9802 is greater than 0.01. We accept H0 and % of no drivers admiting to drive while drowsly is less than 51%

In [12]:
from scipy.stats import norm
import pandas as pd

def propotion(pro):
    pbar = pro/500
    pzero = 0.51 #hypothesised value
    n =500
    mu,sigma = 0,1
    
    z1 = pbar - pzero
    z2 = (pzero*(1-pzero)/n)**0.5

    z = z1/z2
    
    #The are on the left side of STD, NOR
    p = norm.cdf(z,mu,sigma)

    data = dict()
    data['pbar'] = pro
    data['z'] = z
    data['Value of p']= 1-p
    
    if (1 -p ) < 0.01:
        data['Accept'] = 'H1'
    else :
        data['Accept'] = 'H0'
    
    return data 


df100 = pd.DataFrame({'pbar':[],'z':[],'Value of p': [],'Accept': []})
for pro in range(232,501):
    df100= df100.append(propotion(pro),ignore_index = True)
    
    


In [13]:
df100

pbar          z  Value of p Accept
0    232.0  -2.057594    0.980185     H0
1    233.0  -1.968133    0.975474     H0
2    234.0  -1.878673    0.969855     H0
3    235.0  -1.789212    0.963210     H0
4    236.0  -1.699752    0.955411     H0
..     ...        ...         ...    ...
264  496.0  21.560008    0.000000     H1
265  497.0  21.649468    0.000000     H1
266  498.0  21.738929    0.000000     H1
267  499.0  21.828390    0.000000     H1
268  500.0  21.917850    0.000000     H1

[269 rows x 4 columns]

In [14]:
from tabulate import tabulate
print(tabulate(df100,headers='keys',tablefmt="github"))

|     |   pbar |          z |   Value of p | Accept   |
|-----|--------|------------|--------------|----------|
|   0 |    232 | -2.05759   |  0.980185    | H0       |
|   1 |    233 | -1.96813   |  0.975474    | H0       |
|   2 |    234 | -1.87867   |  0.969855    | H0       |
|   3 |    235 | -1.78921   |  0.96321     | H0       |
|   4 |    236 | -1.69975   |  0.955411    | H0       |
|   5 |    237 | -1.61029   |  0.946333    | H0       |
|   6 |    238 | -1.52083   |  0.935849    | H0       |
|   7 |    239 | -1.43137   |  0.923838    | H0       |
|   8 |    240 | -1.34191   |  0.910187    | H0       |
|   9 |    241 | -1.25245   |  0.894797    | H0       |
|  10 |    242 | -1.16299   |  0.877583    | H0       |
|  11 |    243 | -1.07353   |  0.858483    | H0       |
|  12 |    244 | -0.984067  |  0.837459    | H0       |
|  13 |    245 | -0.894606  |  0.814501    | H0       |
|  14 |    246 | -0.805146  |  0.789632    | H0       |
|  15 |    247 | -0.715685  |  0.762907    | H0 